### ISSUE TITLE
Method invocation failed because [System.Object[]] does not contain a method named 'Substring'.

### ISSUE THREAD

**Issue by JonasGroeger** [majkinetor/au#20]
Hey!

I am building a chocolatey package on appveyor and get the following errors:

```
.\au\update_all.ps1
Updating 1 automatic packages at 8/31/2016 7:55:58 PM
Method invocation failed because [System.Object[]] does not contain a method named 'Substring'.
At C:\Users\appveyor\Documents\WindowsPowerShell\Modules\AU\2016.8.15\Update-AUPackages.ps1:90 char:21
+ ...             $i.RemoteVersion = ($i.Result -match '^remote version: .+ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : MethodNotFound
 
Cannot convert value "" to type "System.Version". Error: "Version string portion was too short or too long."
At C:\Users\appveyor\Documents\WindowsPowerShell\Modules\AU\2016.8.15\Update-AUPackages.ps1:94 char:36
+ ... rsion = if ([version]$i.RemoteVersion -gt [version]$i.NuspecVersion)  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvalidCastParseTargetInvocation
 
   pycharm-professional has no updates
```

Is this something that breaks something?

Log: https://ci.appveyor.com/project/Jonas/chocolatey-packages/build/9

**Comment by majkinetor** [comment]
I didn't check it out very carefully, but I see at least one error: Checksum needs to be Checksum32. 

Also, did you run the updater on your own machine ?

**Comment by majkinetor** [comment]
The 'Checksum' error would actually not make an error - it would simply not verify your checksum against executable but would calculate it (and you already parsed it from the site) so the entire downloading is usless. This however, isn't your problem.

Your packages seems to work OK when I run it (I disabled automatic checksum):

```
> updateall py* @{Force = $true } | set x
> $x.Result
pycharm-professional - checking updates
nuspec version: 2016.2.2.20160901
remote version: 2016.2.2
No new version found, but update is forced
Automatic checksum skipped
Updating files
  $Latest data:
    Checksum        (String)    a49699f83936a04d1294a697aef249940709406e29afc6427c3eca81495ae51b
    NuspecVersion   (String)    2016.2.2.20160901
    PackageName     (String)    pycharm-professional
    URL             (String)    https://download.jetbrains.com/python/pycharm-professional-2016.2.2.exe
    Version         (String)    2016.2.2

  pycharm-professional.nuspec
    updating version using Chocolatey fix notation: 2016.2.2.20160901 -> 2016.2.2.20160901
  tools\ChocolateyInstall.ps1
    (^[$]downloadUrl\s*=\s*)('.*') = $1'https://download.jetbrains.com/python/pycharm-professional-2016.2.2.exe'
    (^[$]checksum\s*=\s*)('.*') = $1'a49699f83936a04d1294a697aef249940709406e29afc6427c3eca81495ae51b'
Chocolatey v0.10.0
Attempting to build package from 'pycharm-professional.nuspec'.
Successfully created package 'C:\Work\_test\chocolatey-packages\automatic\pycharm-professional\pycharm-professional.2016.2.2.20160901.nupkg'
Package updated
```

I will look into some more. It would be helpful to provide `update_info.xml` artifact in your appveyor so I can examine the complete output (see https://github.com/majkinetor/chocolatey/blob/master/appveyor.yml#L42).

**Comment by majkinetor** [comment]
And yes, this is broken. In your log `RemoteVersion` is not shown which is strange because package then should fail.

**Comment by majkinetor** [comment]
I encountered a situation similar to this one that turns out to be caused by the bug in `Invoke-WebRequest` which I reported to Microsoft [here](https://windowsserver.uservoice.com/forums/301869-powershell/suggestions/15692703-invoke-webrequest-returned-object-hangs-when-acces). Perhaps that is happening here.

It turns out that `iwr` hangs sometimes so thread gets killed by the AU and there is no 'remote-version' string in the output (the error you provided happens in such case). 

**Adding or removing** `UseBasicParsing` option to `iwr`  to helps almost always.
